### PR TITLE
[chore] ambient modules and imports

### PR DESCRIPTION
**Pull request by ignatiusmb**
A stepping stone for #2223 and a groundwork for #2249

### CODE REVIEW

**Comment by ignatiusmb** [comment]
We can bikeshed `ambient-imports.d.ts` and `ambient-modules.d.ts`, but maybe we'll just have it in one file for this PR now to prevent the scrambled diffing on ambient-modules.

File: packages/adapter-cloudflare-workers/index.js [hunk 1]
```diff
@@ -8,15 +8,11 @@ import { fileURLToPath } from 'url';
...
  */
 
-/**
- * @param {{
- *   esbuild?: (defaultOptions: BuildOptions) => Promise<BuildOptions> | BuildOptions;
- * }} [options]
- **/
+/** @type {import('.')} */
 export default function (options) {
-	/** @type {import('@sveltejs/kit').Adapter} */
```
Comment (benmccann) [inline packages/adapter-cloudflare-workers/index.js:17]: I'm not sure about removing this. When I open up `adapter-static` in VS Code in `master` it complains: 
>Binding element 'utils' implicitly has an 'any' type

File: packages/adapter-node/src/server.js [hunk 2]
```diff
@@ -28,9 +30,9 @@ export function createServer({ render }) {
 
 	const assets_handler = fs.existsSync(paths.assets)
 		? sirv(paths.assets, {
-				setHeaders: (res, pathname, stats) => {
-					// eslint-disable-next-line no-undef
-					if (pathname.startsWith(APP_DIR)) {
+				setHeaders: (res, pathname) => {
+					// @ts-ignore - dynamically replaced with define
```
Comment (benmccann) [inline packages/adapter-node/src/server.js:34]: we should probably use `@ts-expect-error`

File: packages/adapter-node/src/index.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
 // TODO hardcoding the relative location makes this brittle
-import { init, render } from '../output/server/app.js'; // eslint-disable-line import/no-unresolved
-import { path, host, port } from './env.js'; // eslint-disable-line import/no-unresolved
+// @ts-ignore
```
Comment (benmccann) [inline packages/adapter-node/src/index.js:2]: any idea why this is needed here, but not in `adapter-cloudflare-workers/files/entry.js`?

Comment (ignatiusmb) [reply packages/adapter-node/src/index.js:2]: I was also scratching my head for a while trying to figure out this and ended up just ignoring this one, turns out the tsconfig for all the other serverless adapters weren't including their "source" files as well

This should just be a reminder though and will hopefully be resolved in the upcoming PR

**Comment by ignatiusmb** [comment]
Reminder for the next PR to also resolve #625

File: packages/adapter-node/src/server.js [hunk 4]
```diff
@@ -40,6 +42,7 @@ export function createServer({ render }) {
 		: noop_handler;
 
 	const server = polka().use(
+		// @ts-ignore TODO - compression return doesn't play well with polka.use
```
Comment (benmccann) [inline packages/adapter-node/src/server.js:45]: see https://github.com/lukeed/polka/issues/173. maybe the solution is to cast it to `unknown` and then to `polka.Middleware`?  either that or I guess someone would need to dig into to understand what is different about the Polka types

File: packages/kit/src/runtime/client/start.js [hunk 5]
```diff
@@ -1,7 +1,7 @@
 // @ts-ignore - value will be replaced on build step
-import Root from 'ROOT'; // eslint-disable-line import/no-unresolved
+import Root from 'ROOT';
```
Comment (benmccann) [inline packages/kit/src/runtime/client/start.js:2]: we should probably switch these two to `@ts-expect-error`

**Comment by ignatiusmb** [comment]
Updated the comments. For the polka issue we'll just have to ignore them for now, casting the types would make sense if we're actually writing TS, but ignoring them makes more sense here, plus we've also put our reasons for every ignore and expect errors so it's clearer than casting (which we should try to avoid if possible).

**Review by dummdidumm — COMMENTED** [review]
I don't know the reasons for adding `@sveltejs/kit/node` etc, so I can't say much about that. Type-wise, this looks good. If I remember correctly, the `type Page = import('@sveltejs/kit').Page` were needed because else they were not picked up during development on Kit itself. Did that change?

**Comment by ignatiusmb** [comment]
Thanks, and yes! Other than having to disable eslint duplicate imports, it seems to be working fine now.

**Review by dummdidumm — APPROVED** [review]

**Review by benmccann — APPROVED** [review]

**Pull request merged**